socketcan_bridge/src/socketcan_recover_ctrl.cpp

```diff
@@ -0,0 +1,81 @@
+#include <socketcan_bridge/socketcan_recover_ctrl.h>
+
+namespace socketcan_bridge
+{
+    
+    /**
+     * @brief Constructor for the recovery control class
+     *
+     * Initializes the publisher, and sets up a timer to periodically check the bus state
+     */
+    SocketCANRecoverCtrl::SocketCANRecoverCtrl(ros::NodeHandle* nh, ros::NodeHandle* nh_param,
+        boost::shared_ptr<can::DriverInterface> driver) : driver_(driver)
+    {
+        state_pub_ = nh->advertise<can_msgs::CanState>("can_state", 1, true);
+
+        timer_ = nh->createWallTimer(ros::WallDuration(5), [this](const ros::WallTimerEvent& event) {recover();}, true, false);
+
+        state_listener_ = driver_->createStateListener(
+            can::StateInterface::StateDelegate(this, &SocketCANRecoverCtrl::stateCallback));
+
+    }
+
+    /**
+     * @brief Checks the state of the bus, if !statie.isReady() then the
+     * recover timer is started to fire in 5secs, otherwise we stop the timer
+     */
+    void SocketCANRecoverCtrl::stateCallback(const can::State & state) {
+        publishStatus(state);
+        if(!state.isReady())
+        {            
+            timer_.start();
+        }
+        else
+        {
+            timer_.stop();
+        }
+    }
+
+    /**
+     * @brief Recover the bus from an error state
+     *
+     * Calls the driver's recover() function
+     */
+    void SocketCANRecoverCtrl::recover() 
+    {
+        timer_.stop();
```

Comment:
no need to stop here

Reply:
This stop is explicitly required, you will have to forgive me I don't remember why, either it made the timer hang holding a reference preventing from being able to simple Ctrl-C the app, or it was needed to call start below, ill look into my reasoning here

Reply:
Ah yes, it might block at ctrl+c.
But fixing it here is just handling the symptom.